### PR TITLE
Add Application Gateway OAuth tests

### PR DESCRIPTION
**Pull request by franpog859**
**Description**

Changes proposed in this pull request:

- Add Application Gateway OAuth tests

**Related issue(s)**

https://github.com/kyma-project/kyma/issues/14654

### CODE REVIEW

File: tests/components/application-connector/internal/testkit/test-api/apis.go [hunk 1]
```diff
@@ -11,13 +11,20 @@ import (
...
 	router := mux.NewRouter()
 
 	router.HandleFunc("/v1/health", alwaysOk).Methods("GET")
 	api := router.PathPrefix("/v1/api").Subrouter()
 	api.Use(Logger(logOut, logger.DevLoggerType))
 
+	oauth := NewOAuth(oAuthCredentials.ClientID, oAuthCredentials.ClientSecret)
+
+	{
+		api.HandleFunc("/oauth/token", oauth.Token).Methods(http.MethodPost)
```
Comment (VOID404) [inline tests/components/application-connector/internal/testkit/test-api/apis.go:24]: there is already an `/oauth` block, starting from line 40

File: tests/components/application-connector/internal/testkit/test-api/oauth.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+const (
+	clientIDKey     = "client_id"
+	clientSecretKey = "client_secret"
+	grantTypeKey    = "grant_type"
+)
+
+type OauthResponse struct {
+	AccessToken string `json:"access_token"`
+	TokenType   string `json:"token_type"`
+	ExpiresIn   int    `json:"expires_in"`
```
Comment (VOID404) [inline tests/components/application-connector/internal/testkit/test-api/oauth.go:28]: This field [is not required](https://www.rfc-editor.org/rfc/rfc6749#section-4.2.2) and we do not handle it correctly, so maybe we should omit it

File: tests/components/application-connector/internal/testkit/test-api/oauth.go [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+	Scope       string `json:"scope"`
```
Comment (VOID404) [inline tests/components/application-connector/internal/testkit/test-api/oauth.go:29]: This field [is not required](https://www.rfc-editor.org/rfc/rfc6749#section-4.2.2) and we do not handle it correctly, so maybe we should omit it

File: tests/components/application-connector/internal/testkit/test-api/oauth.go [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+			splitToken := strings.Split(authHeader, "Bearer")
+			if len(splitToken) != 2 {
+				handleError(w, http.StatusForbidden, "Bearer token missing")
+				return
+			}
+
+			token := strings.TrimSpace(splitToken[1])
+
+			oh.mutex.RLock()
+			_, found := oh.tokens[token]
```
Comment (VOID404) [inline tests/components/application-connector/internal/testkit/test-api/oauth.go:106]: If we ignore values in the map, maybe we could have `interface{}` and `nil` inside, to signify the lack of meaning of the stored value?

**Review by VOID404 — CHANGES_REQUESTED** [review]

File: tests/components/application-connector/internal/testkit/test-api/apis.go (hunk 1 above)
Comment (franpog859) [reply tests/components/application-connector/internal/testkit/test-api/apis.go:24]: I wanted to separate the token endpoints and the _target url endpoints_ as the token ones will be used across also the CSRF test but the _target urls_ will be different. In this case, there will be shared token urls and separated _target urls_ for each test case

Comment (VOID404) [reply tests/components/application-connector/internal/testkit/test-api/apis.go:24]: We could consider naming endpoint blocks in the future 🤔

**Review by VOID404 — APPROVED** [review]

**Pull request merged**